rotkehlchen/chain/ethereum/modules/balancer/typing.py

```diff
@@ -0,0 +1,86 @@
+from dataclasses import dataclass
+from typing import Any, DefaultDict, Dict, List, NamedTuple, Set, Union
+
+from rotkehlchen.accounting.structures import Balance
+from rotkehlchen.assets.asset import EthereumToken
+from rotkehlchen.assets.unknown_asset import UNKNOWN_TOKEN_KEYS, UnknownEthereumToken
+from rotkehlchen.chain.ethereum.trades import AMMSwap, AMMTrade
+from rotkehlchen.constants import ZERO
+from rotkehlchen.fval import FVal
+from rotkehlchen.typing import ChecksumEthAddress, Price
+
+BALANCER_TRADES_PREFIX = 'balancer_trades'
+
+
+@dataclass(init=True, repr=True)
+class BalancerPoolToken:
+    token: Union[EthereumToken, UnknownEthereumToken]
+    total_amount: FVal
+    user_balance: Balance
+    weight: FVal
+    usd_price: Price = Price(ZERO)
+
+    def serialize(self) -> Dict:
+        serialized_token: Union[str, Dict[str, Any]]
+        if isinstance(self.token, EthereumToken):
+            serialized_token = self.token.serialize()
+        elif isinstance(self.token, UnknownEthereumToken):
+            serialized_token = self.token.serialize_as_dict(keys=UNKNOWN_TOKEN_KEYS)
+        else:
+            raise AssertionError(f'Unexpected type: {type(self.token)}')
+
+        return {
+            'token': serialized_token,
+            'total_amount': self.total_amount,
+            'user_balance': self.user_balance.serialize(),
+            'usd_price': self.usd_price,
+            'weight': self.weight,
+        }
+
+
+@dataclass(init=True, repr=True)
+class BalancerPool:
+    address: ChecksumEthAddress
+    tokens: List[BalancerPoolToken]
+    total_amount: FVal
+    user_balance: Balance
+
+    def serialize(self) -> Dict[str, Any]:
+        return {
+            'address': self.address,
+            'tokens': [token.serialize() for token in self.tokens],
+            'total_amount': self.total_amount,
+            'user_balance': self.user_balance.serialize(),
+        }
+
+
+AddressBalances = Dict[ChecksumEthAddress, List[BalancerPool]]
+DDAddressBalances = DefaultDict[ChecksumEthAddress, List[BalancerPool]]
+TokenPrices = Dict[ChecksumEthAddress, Price]
+
+
+class ProtocolBalance(NamedTuple):
+    address_balances: AddressBalances
+    known_tokens: Set[EthereumToken]
+    unknown_tokens: Set[UnknownEthereumToken]
+
+
+AddressSwaps = Dict[ChecksumEthAddress, List[AMMSwap]]
```

Comment:
I will rename it. It is fine for me to add the `to` in between here and in the other ones if it makes it easier to understand that it is a map. I'm just following the convention we agreed in our last (and I'd say the only proper one)[ naming convention conversation](https://github.com/rotki/rotki/pull/2138#discussion_r560209112).
```
I will stop repeating it in the rest of the review as it seems to be in many places. But just follow the mapping naming scheme I had also mentioned in previous PRs. address_to_swaps here
```

Reply:
That was for what I understood is a list at the time. It's really hard to keep track of all review comments when PRs are huge.

In any case for mappings we need a `_to_` or a `_mapping` or something. I prefer `x_to_y`.

If you can at some point add those naming conventions in the contribution for developer docs. It would be really cool to have them written down. A PR comment is definitely not enough.